Post/tests.py

```diff
@@ -1,13 +1,45 @@
 from Post.models import Post
+from django.db.models.query import QuerySet
+import pytest
 
 
-class TestPost():
+@pytest.mark.django_db
+def test_post_str(get_post):
 
-    def test_post_creation(self):
+    assert str(get_post) == "Shoval traveled The Dead Sea and wrote: beautiful place"
 
-        post1 = Post(nameOfPoster='Shoval',
-                     nameOfLocation='The Dead Sea',
-                     photoURL='www.testPost.com',
-                     Description='beautiful place')
 
-        assert str(post1) == "Shoval traveled The Dead Sea and wrote: beautiful place"
+@pytest.mark.django_db
+def test_post_creation(get_post):
+    get_post.save()
+
+    assert Post.objects.filter(pk=get_post.id).exists()    # checks if the post is saved
+    assert Post.objects.get(pk=get_post.id) == get_post       # check if the post saved correctly
+
+
+@pytest.mark.django_db
+def test_post_delete(get_post):
+    get_post.save()
+
+    assert Post.objects.filter(pk=get_post.id).exists()
+    Post.objects.get(pk=get_post.id).delete()
+    assert Post.objects.filter(pk=get_post.id).exists() is False
+
+
+@pytest.mark.django_db
+def test_all_posts():
+
+    posts = Post.all_posts()
+
+    assert isinstance(posts, QuerySet)
+    assert all(isinstance(post, Post) for post in posts)
+
```

Comment:
why don't you add assertion of the values as Barak did in his example? using the value_list of the queryset?

Reply:
Consider asserting posts are not None and len > 0

Reply:
@maayanhd, I tried to do that, but in this queryset there are many values and I think we will need to change the test every time we add a new post. If you have an idea how to add assertation for the values that will be useful - I will be happy to hear that!

Reply:
@Yarboa, I added the changes you asked for.